metabase/metabase
# fix time series chart is broken due to yyyymmdd numeric x-axis format

Closes https://github.com/metabase/metabase/issues/41289

### Description

Fixes time series charts when for their x-values column contains numeric values in yyyymmdd format like `19210413` as numbers.
We've been having our own parse timestamp function that uses moment `parseTimestamp` in `frontend/src/metabase/lib/time.ts`. Since we want to move away from moment and use dayjs all the code for the cartesian charts was using dayjs. So this PR introduces and uses the new `parseTimestamp` function based on dayjs. It required to include a missing in dayjs plugin `parseZone` which was implemented by the community but not yet got merged.

### How to verify

Ensure the following x values are parsed as 13 April 1921 and 13 April 2021:
```sql
select 19210413 x, 10 y
union all select 20210413 x, 11 y
```

### Demo

<img width="1724" alt="Screenshot 2024-04-17 at 7 35 10 PM" src="https://github.com/metabase/metabase/assets/14301985/18d76f41-8f40-47c0-8df1-a872f85b4da6">

### Checklist
